src/modules/alerting.ts

```diff
@@ -42,20 +52,30 @@ class AlertingApi extends BaseApi {
     /**
      * Displays a toast popup.
      *
+     * Permanent/Sticky toasts are not allowed.  Therefore, toasts must specify either a manual
+     * dismissal (`showCloseButton: true`) or an automatic dismissal (`timeout >= 0`). Both
+     * `showCloseButton` and `timeout` can be specified to provide both dismissal options.
+     *
+     * Error toasts (`type: 'error'`) require manual dismissal and must be specified with `showCloseButton: true`.
+     * Any specified `timeout` will be ignored.
+     *
      * ```ts
      * myClientApp.alerting.showToastPopup("Hello world", "Hello world, how are you doing today?");
      * ```
      *
      * ```ts
      * var options = {
```

Comment:
Updated our examples while I was here to reflect our newly enforced contract (at least for TS consumers)